### PR TITLE
Refactored the integration tests to take the workflow inputs

### PR DESCRIPTION
**Pull request by nathalapooja**
*Description of changes:*
 1. Separated the Beta cluster and GA cluster integration tests.
 2. Generated test matrix for different k8s versions.
 3. Added workflow inputs for addon name and version.

Succeeded Tests: https://github.com/aws/amazon-cloudwatch-agent-operator/actions/runs/6656933801

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by sky333999** [comment]
Mind making the PR title a bit more descriptive?

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -1,22 +1,23 @@
 # Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-name: Run Integration Tests
+name: Run EKS addon Integration Tests
```
Comment (sky333999) [inline .github/workflows/integration-test.yml:4]: nit: mind renaming the file itself? Perhaps `eks-add-on-integ-test` and `eks-beta-add-on-integ-test`?

File: .github/workflows/integration-test.yml [hunk 2]
```diff
@@ -46,42 +44,24 @@ jobs:
...
-            terraform init
-            if terraform apply --auto-approve; then
-              terraform destroy -auto-approve
-            else
-              terraform destroy -auto-approve && exit 1
-            fi
+      - name: Generate matrix
+        id: set-matrix
+        run: |
+         echo "::set-output name=eks_addon_matrix::$(echo $(cat integration-tests/generator/k8s_versions_matrix.json))"
```
Comment (sky333999) [inline .github/workflows/integration-test.yml:50]: We dont really have a generator.. but its fine for now.

**Review by sky333999 — DISMISSED** [review]

File: .github/workflows/integration-test.yml (hunk 1 above)
Comment (nathalapooja) [reply .github/workflows/integration-test.yml]: Updated in revision

**Review by dchappa — APPROVED** [review]

**Review by sky333999 — APPROVED** [review]

**Pull request merged**